src/test/java/com/productdock/library/inventory/book/BookServiceShould.java

```diff
@@ -0,0 +1,73 @@
+package com.productdock.library.inventory.book;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.productdock.library.inventory.producer.Publisher;
+import com.productdock.library.inventory.record.RentalRecordMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import static com.productdock.library.inventory.data.provider.BookMother.*;
+import static com.productdock.library.inventory.data.provider.RentalRecordMother.defaultRentalRecord;
+import static com.productdock.library.inventory.data.provider.RentalRecordMother.defaultRentalRecordMessage;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceShould {
+
+    @InjectMocks
+    private BookService bookService;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private Publisher publisher;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @Mock
+    private RentalRecordMapper rentalRecordMapper;
+
+    @BeforeEach
+    final void before() {
+        bookRepository.deleteAll();
+    }
+
+    @Test
+    void updateBookState_whenBookExists() throws Exception {
```

Comment:
I must quit review now because my cell phone battery is dead, will continue soon :)